src/sql/workbench/services/connection/browser/connectionDialogWidget.ts

```diff
@@ -127,10 +127,9 @@ export class ConnectionDialogWidget extends Modal {
 			}
 		}
 
-		this._providerTypeSelectBox.setOptions(filteredProviderDisplayNames.filter((providerDisplayName, index) =>
-			// Remove duplicate listings (CMS uses the same display name)
-			filteredProviderDisplayNames.indexOf(providerDisplayName) === index)
-		);
+		// Remove duplicate listings (CMS uses the same display name)
+		let dedupMap = this._connectionManagementService.getDedupeConnectionProvidersByNameMap(this.providerNameToDisplayNameMap);
+		this._providerTypeSelectBox.setOptions(Object.keys(dedupMap).map(k => dedupMap[k]));
```

Comment:
Hm, what's the type of ```dedupMap```? Why do you need to ```map``` it again? Can't we just return the map from that ```get...``` method?

Reply:
dedupMap returns the object of type { [providerDisplayName: string]: string } and we only need display names in the box. But I am thinking to rather return a map. This object dictionary interface is making things super complicated.

Reply:
I see, yeah, I thought it was actually generating a ```Map```... Can't you just do ```Object.values``` if that's what you need?

Reply:
Nope, this is an object in typescript and you cannot access values directly. I am trying to see if I should return a Map instead as we will only be using this object for iteration/lookup. It will make accessing keys and values easier.
Looking into this: https://medium.com/front-end-weekly/es6-map-vs-object-what-and-when-b80621932373